### PR TITLE
Die if sequins is ZK-flapping

### PR DESCRIPTION
**Pull request by vasi-stripe**
Our service runner should then restart sequins, allowing us to successfully reconnect. Yes, this is ugly :(

Also add some extra debugging info to our ZK watcher.

### CODE REVIEW

File: sequins.go [hunk 1]
```diff
@@ -216,6 +221,12 @@ func (s *sequins) initCluster() error {
 		return err
 	}
 
+	flapNotify := zkWatcher.SetFlapThreshold(flapMax, flapDuration)
+	go func() {
+		<-flapNotify
+		panic("Dying due to ZK flapping")
```
Comment (scottjab-stripe) [inline sequins.go]: I really don't like panics as they are hard to parse in splunk, can we change this to a `log.Fatal`.  We don't really care about the stack trace here do we?

File: zk/watcher_test.go [hunk 2]
```diff
@@ -127,3 +130,46 @@ func TestZKRemoveWatch(t *testing.T) {
...
+
+	for i := 0; i < 5; i++ {
+		simulateError(w)
+	}
+	select {
+	case <-time.After(100 * time.Millisecond):
+		assert.Fail(t, "Flaps in a short period should notify")
+	case <-flapNotify:
+	}
+	println("DONE")
```
Comment (scottjab-stripe) [inline zk/watcher_test.go]: Is this print needed?

**Comment by vasi-stripe** [comment]
Fixed those nits. PTAL @scottjab-stripe

**Comment by scottjab-stripe** [comment]
LGTM
